_tests/test_validate.py

```diff
@@ -62,143 +64,449 @@ def test_checkDownloadUrlFormat_validExampleURL(self):
 			ADDON_PACKAGE,
 			expectedSha='abc'
 		)
 		errors = list(errors)
-		expectedErrorMessage = validate.ValidationErrorMessage.CHECKSUM_FAILURE.value
-		actualSha = self.submissionData["sha256"].lower()
 		self.assertEqual(
 			errors,
-			[expectedErrorMessage.format(actualSha)]
+			[f"Sha256 of .nvda-addon at URL is: {self.validSha.lower()}"]
 		)
 
-	def test_checkSummaryMatchesDisplayName_valid(self):
+
+class Validate_checkSummaryMatchesDisplayName(unittest.TestCase):
+	def setUp(self):
+		self.submissionData = getValidAddonSubmission()
+		self.manifest = getAddonManifest()
+
+	def tearDown(self):
+		self.submissionData = None
+		self.manifest = None
+
+	def test_valid(self):
 		errors = list(
 			validate.checkSummaryMatchesDisplayName(self.manifest, self.submissionData)
 		)
 		self.assertEqual(errors, [])
 
-	def test_checkSummaryMatchesDisplayName_invalid(self):
+	def test_invalid(self):
 		badDisplayName = "bad display Name"
 		self.submissionData["displayName"] = badDisplayName
 		errors = list(
 			validate.checkSummaryMatchesDisplayName(self.manifest, self.submissionData)
 		)
-		expectedErrorMessage = validate.ValidationErrorMessage.NAME.value
 		self.assertEqual(
 			errors,
-			[expectedErrorMessage.format(self.manifest["summary"], badDisplayName)]
+			[
+				f"Submission 'displayName' must be set to '{self.manifest['summary']}' in json file."
+				f" Instead got: '{badDisplayName}'"
+			]
 		)
 
-	def test_checkDescriptionMatches_valid(self):
+
+class Validate_checkDescriptionMatches(unittest.TestCase):
+	def setUp(self):
+		self.submissionData = getValidAddonSubmission()
+		self.manifest = getAddonManifest()
+
+	def tearDown(self):
+		self.submissionData = None
+		self.manifest = None
+
+	def test_valid(self):
 		errors = list(
 			validate.checkDescriptionMatches(self.manifest, self.submissionData)
 		)
 		self.assertEqual(errors, [])
 
-	def test_checkDescriptionMatches_invalid(self):
+	def test_invalid(self):
 		badDesc = "bad description"
 		self.submissionData["description"] = badDesc
 		errors = list(
 			validate.checkDescriptionMatches(self.manifest, self.submissionData)
 		)
-		expectedErrorMessage = validate.ValidationErrorMessage.DESC.value
 		self.assertEqual(
 			errors,
-			[expectedErrorMessage.format(self.manifest["description"], badDesc)]
+			[
+				f"Submission 'description' must be set to '{self.manifest['description']}' in json file."
+				f" Instead got: '{badDesc}'"
+			]
 		)
 
-	def test_checkNameMatchesPath_valid(self):
+
+class Validate_checkAddonId(unittest.TestCase):
+	"""
+		Manifest 'name' considered source of truth for addonID
+		Must match:
+		- Submission file name '<addonID>/<version>.json'
+		- `addonId` within the submission JSON data
+	"""
+	def setUp(self):
+		self.submissionData = getValidAddonSubmission()
+		self.manifest = getAddonManifest()
+
+	def tearDown(self):
+		self.submissionData = None
+		self.manifest = None
```

Comment:
Could this repeated setUp/tearDown code be inherited from a base TestCase across all these TestCases?

Reply:
Possibly. I'd prefer not inheritance for code reuse, the tests don't necessarily have the same dependencies. It's pretty minor repetition. On the plus side it keeps the behavior of a related group of tests local.